nuclio_sdk/platform.py

```diff
@@ -13,20 +13,51 @@
 # limitations under the License.
 
 import json
+import signal
 import http.client
 
 import nuclio_sdk
 import nuclio_sdk.helpers
 
 
 class Platform(object):
-    def __init__(self, kind, namespace="default", connection_provider=None):
+    def __init__(
+        self,
+        kind,
+        namespace="default",
+        connection_provider=None,
+        on_control_callback=None,
+    ):
         self.kind = kind
         self.namespace = namespace
 
         # connection_provider is used for unit testing
         self._connection_provider = connection_provider or http.client.HTTPConnection
 
+        self._control_callback = on_control_callback
+
+    async def explicit_ack(self, event):
+        """
+        Ensures marking the offset on the stream according to the given arguments
+
+        :param event
+        :type Event
+        """
+        message = event.compile_explicit_ack_message()
+        if self._control_callback:
+            await self._control_callback(message)
+        else:
+            raise Exception(
+                "Cannot send explicit ack since control callback was not initialized"
+            )
+
+    def on_signal(self, callback, sig=signal.SIGTERM):
+        """
+        Registers the callback to signal.
+        this function should be called from init_context.
+        """
+        signal.signal(sig, callback)
```

Comment:
Also, to support asynchronous callback upon demand it is possible to use the event loop to handle signal catching. Perhaps you want to support that as well

Reply:
You mean registering to the signal using the event loop, or calling the `callback` using the loop?

I tried to avoid passing the loop itself from the wrapper to the Platform class to make it a bit more simple.